CMakeLists.txt

```diff
@@ -8,6 +8,7 @@ option(WITH_EXAMPLES "Whether to include examples" ON)
 option(WITH_TESTS "Whether to include tests" ON)
 
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
In which cases do we need the compile database? Is it to have access to the compiler options?

Reply:
This is mostly for tooling support. For instance I use clangd for completion in vim and it uses the compilation database.

QtCreator, vscode, and other tools also make use of it.
We could make the default off if you want.

Reply:
There is some contention on what is the right approach and there are undocumented features of clangd regarding where it looks, see https://github.com/clangd/clangd/issues/770.

Common approaches are: copying, making a symlink in the root, configuring a path in a .clangd file at the root.

They all require creating either files or links and adding them to .gitignore so, at least from my experience, simply copying the file looks cleaner.

In any case I made it an option that is off by default to avoid creating files the user does not need.